umibukela/static/javascript/charts.js

```diff
@@ -322,47 +322,77 @@ var PrintMaterials = function() {
+      // Create data objects for each available period
+      for(var i=0;i < periodKeys.length;i++) {
+        var periodName = periodKeys[i];
+        maleData.push({ period: periodName, year: periodName == 'current' ? years[1] : years[0] });
+        femaleData.push({ period: periodName, year: periodName == 'current' ? years[1] : years[0] });
+      }
 
-        maleData.push({ period: period, year: period == 'current' ? years[1] : years[0] });
-        femaleData.push({ period: period, year: period == 'current' ? years[1] : years[0] });
+      var compare = function(a, b) {
+        if (a.current.key == "positive")
+          return 1;
+        if (b.current.key == "negative")
+          return 1;
+        if (b.current.key == "positive")
+          return -1;
+        if (a.current.key == "negative")
+          return -1;
+        if (a.current.key == "yes")
+          return 1;
+        if (b.current.key == "no")
+          return 1;
+        if (b.current.key == "yes")
+          return -1;
+        if (a.current.key == "no")
+          return -1;
+        if (b.current.key == "very_difficult")
+          return 1;
+        if (a.current.key == "easy")
+          return 1;
+        if (a.current.key == "very_difficult")
+          return -1;
+        if (b.current.key == "easy")
+          return -1;
+        console.log("unable to provide ordering: ", a.current.key, b.current.key);
       }
+      responses.sort(compare);
 
+      // Set the category values for the objects in the D3 data arrays
+      // Also add labels to the legend label array
       responses.forEach(function(response, i) {
-        for(period in response) {
-          var male_datum = _.find(maleData, function(item) { return item.period == period });
-          var female_datum = _.find(femaleData, function(item) { return item.period == period });
-          var key = response[period].key;
-          var label = response[period].label;
+        for(var i2 = 0; i2 < options.periodKeys.length; i2++) {
+          var periodName = options.periodKeys[i2];
+          var male_datum = _.find(maleData, function(item) { return item.period == periodName });
+          var female_datum = _.find(femaleData, function(item) { return item.period == periodName });
+          var key = response[periodName].key;
+          var label = response[periodName].label;
 
           if(key) labels.push(key);
           if(!_.contains(legendLabels, label)) legendLabels.push(label);
 
-          male_datum[labels[i]] = response[period].count.male;
-          female_datum[labels[i]] = response[period].count.female;
+          male_datum[labels[i]] = response[periodName].count.male;
+          female_datum[labels[i]] = response[periodName].count.female;
         }
       });
 
       // Range depends on number of response types
       var zRange = legendType == 'yes/no' ? [self.ORANGE,self.BLACK] : [self.ORANGE,self.WHITE,self.BLACK];
 
-      if (legendType == 'yes/dk/no') {
```

Comment:
I replaced yes/dk/no and yes/no/na with yes/other/no everywhere now because we don't need to do label order hackery when we don't make assumptions about the order they're "coming in" - we use keys to explicitly arrange them for our posters.